### PR TITLE
do not need to continue after we have found the version

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
do not need to continue after we have found the version

## Motivation and Context
it looks like we must break in this path

## How to test this PR?
Not sure has been around for a while.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/xl-storage-format-v2.go [hunk 1]
```diff
@@ -1788,7 +1788,7 @@ func (x xlMetaV2) ToFileInfo(volume, path, versionID string, inclFreeVers, allPa
 			}
 		}
 		if found {
-			continue
+			break
```
Comment (vadmeste) [inline cmd/xl-storage-format-v2.go:1791]: I think we need to continue, for now. It looks like we are counting the number of free versions in this xl.meta so we can report correctly the number of versions in line 1827 (not sure why NumVersions is used, mostly for ILM max non current versions count)

Comment (harshavardhana) [reply cmd/xl-storage-format-v2.go:1791]: Okay then this is a problem @krisis we need to see why this is the case - because this is expensive.

**Review by poornas — APPROVED** [review]

File: cmd/xl-storage-format-v2.go (hunk 1 above)
Comment (klauspost) [reply cmd/xl-storage-format-v2.go:1791]: Not really. It is just checking the flag for each entry. Even with 100K versions that is not really going to be slow.

Comment (harshavardhana) [reply cmd/xl-storage-format-v2.go:1791]: Okay then, let's leave this for now.

**Pull request closed without merging**